scripts/editor/register-blocks.js

```diff
@@ -483,7 +483,14 @@ const registerVariation = (
 ) => {
 
 	// Append globalManifest data in to output.
-	blockManifest['icon'] = getIconOptions(globalManifest, blockManifest);
+	if (blockManifest['icon']) {
+		blockManifest['icon'] = getIconOptions(globalManifest, blockManifest);
+	} else {
+		// There is no icon passed to variation, use it's parent icon instead
+		import(`./../../blocks/init/src/Blocks/custom/${blockManifest.parentName}/manifest.json`).then((parentManifest) => {
+			blockManifest['icon'] = getIconOptions(globalManifest, parentManifest);
+		});
```

Comment:
Consider making the whole method async and then `await`ing the `getIconOptions`.

```suggestion
		const parentManifest = await import(`./../../blocks/init/src/Blocks/custom/${blockManifest.parentName}/manifest.json`);
		blockManifest['icon'] = getIconOptions(globalManifest, parentManifest);
```

Reply:
the PR approach is wrong so I will remove this comment because it needs to be refactored.